docs/TROUBLESHOOTING.md

```diff
@@ -256,8 +261,51 @@ Most likely this is related to a billing quota issue.
 
 try
 
-```
+```bash
 gcloud alpha billing projects link projects/some-project --billing-account XXXXXX-XXXXXX-XXXXXX
 ```
 
 If output states `Cloud billing quota exceeded`, you can use the [Request Billing Quota Increase](https://support.google.com/code/contact/billing_quota_increase) form to request a billing quota increase.
+
+### Terraform Error acquiring the state lock
+
+**Error message:**
+
+```text
+Error: Error acquiring the state lock
+```
+
+**Cause:**
+
+This message means you are trying to execute a Terraform command with the [Terraform State locked](https://www.terraform.io/language/state/locking).
+
+If the Terraform process was unable to finish due to an unexpected event, i.e build timeout or terraform process killed. It will keep the Terraform State **locked**.
+
+**Solution:**
+
+Keep in mind that after removing the Terraform State lock you will need to [review and manipulate](https://www.terraform.io/cli/state#manipulating-terraform-state) your Terraform State and maybe [import](https://www.terraform.io/cli/import#import) some resources.
+
+Here are some sample commands on how to remove the Terraform State lock.
+
+- Update `<YOUR-REPO-DIR>` in the command below to navigate to repo directory of the step you got the error, it should be `gcp-org`, `gcp-environments`, `gcp-networks`, `gcp-projects`, or `bu*-example-app`.
+- Update `<YOUR-FOUNDATION-EXAMPLE-DIR>` in the command below to get the backend state bucket and update `backend.tf` file.
+- Run `terraform init` to initialize terraform with the terraform state from gcs bucket.
+
+```bash
+cd <YOUR-REPO-DIR>
+export backend_bucket=$(terraform -chdir="<YOUR-FOUNDATION-EXAMPLE-DIR>/0-bootstrap/" output -raw gcs_bucket_tfstate)
+for i in `find -name 'backend.tf'`; do sed -i "s/UPDATE_ME/${backend_bucket}/" $i; done
+terraform init
+```
+
+- Run the command below to remove the Terraform State lock, it will ask for a confirmation and you will be able to run terraform commands again.
+
+```bash
+terraform force-unlock $(terraform apply 2>&1 > /dev/null | grep "ID:" | sed -e 's/.*[^0-9]\([0-9]\+\)[^0-9]*$/\1/')
+```
+
+**Notes:**
+
+- The remove Terraform State lock command **does not** change anything in your state.
+- If you got a Terraform State locked error **most of the time** the Terraform State will be inconsistent.
+- It is a **strong recommendation** to review the Terraform State before trying to continue on the step you stopped.
```

Comment:
I would link to upstream docs rather than having this info here
https://www.terraform.io/language/state/locking#force-unlock

Reply:
Done